ddcommon-ffi/src/slice_mut.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use core::slice;
+use serde::ser::Error;
+use serde::Serializer;
+use std::borrow::Cow;
+use std::fmt::{Debug, Display, Formatter};
+use std::hash::{Hash, Hasher};
+use std::marker::PhantomData;
+use std::os::raw::c_char;
+use std::str::Utf8Error;
+
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct MutSlice<'a, T: 'a> {
+    /// Should be non-null and suitably aligned for the underlying type. It is
+    /// allowed but not recommended for the pointer to be null when the len is
+    /// zero.
+    ptr: *mut T,
+
+    /// The number of elements (not bytes) that `.ptr` points to. Must be less
+    /// than or equal to [isize::MAX].
+    len: usize,
+    _marker: PhantomData<&'a mut [T]>,
+}
+
+impl<'a, T: 'a> core::ops::Deref for MutSlice<'a, T> {
+    type Target = [T];
+
+    fn deref(&self) -> &Self::Target {
+        self.as_slice()
+    }
+}
+
+impl<T: Debug> Debug for MutSlice<'_, T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        self.as_slice().fmt(f)
+    }
+}
+
+impl<T: Eq> PartialEq<Self> for MutSlice<'_, T> {
+    fn eq(&self, other: &Self) -> bool {
+        **self == **other
+    }
+}
```

Comment:
Is this required? Deref will pick up a lot of things, will it pick up an eq?

Reply:
Removed